File: isletimsistemicalismasidart-calismalarim/lib/giris_ekrani/repository/KullaniciBilgileri.dart
import 'package:shared_preferences/shared_preferences.dart';

class KullaniciBilgileri {


  late SharedPreferences _sp;

  Future<void> init() async {
    _sp = await SharedPreferences.getInstance();
  }

  bool sifreKaydiOlusturulmusMu() {
    return _sp.getBool("kayitolusturulmusmu") ?? false;
  }

  Future<void> sifreKaydi(String sifre) async {
    await _sp.setString("sifre", sifre);
    await _sp.setBool("kayitolusturulmusmu", true);
  }

  bool sifreSorgulama(String dogrulanacakSifre) {
    return dogrulanacakSifre == (_sp.getString("sifre") ?? "");
  }

  Future<void> guvenlikSorusuKaydet(String guvenlikSorusu) async {
    await _sp.setString("guvenliksorusu", guvenlikSorusu);
  }

  String guvenlikSorusuGetir() {
    return _sp.getString("guvenliksorusu") ?? "";
  }

  Future<void> guvenlikSorusuCevapKaydet(String guvenlikSorusuCevap) async {
    await _sp.setString("guvenliksorusucevap", guvenlikSorusuCevap);
  }

  bool guvenlikSorusuDogrula(String dogrulanacakCevap) {
    return dogrulanacakCevap == (_sp.getString("guvenliksorusucevap") ?? "");
  }

  Future<void> sifreSorulsunMuDegistir(bool sifreSorulsunMu) async {
    await _sp.setBool("sifresorulsunmu", sifreSorulsunMu);
  }

  bool sifreSorulsunMu() {
    return _sp.getBool("sifresorulsunmu") ?? true;
  }

  Future<void> sifreDegistir(String degistirilecekSifre) async {
    await _sp.setString("sifre", degistirilecekSifre);
  }

  Future<void> kullaniciBilgileriSifirla() async {
    await _sp.remove("kayitolusturulmusmu");
    await _sp.remove("sifre");
    await _sp.remove("guvenliksorusu");
    await _sp.remove("sifresorulsunmu");
  }
}

File: isletimsistemicalismasiSwift-main/isletimsistemlericalismasiSwift/isletimsistemlericalismasi/giris_ekrani/repository/KullaniciBilgileri.swift
//
//  KullaniciBilgileri.swift
//  isletimsistemlericalismasi
//
//  Created by Alperen Saraç on 2.01.2025.
//

import Foundation

class KullaniciBilgileri{
    var d:UserDefaults
    
    init() {
        d = UserDefaults.standard
    }
    func sifreKaydiOlusturulmusMu()->Bool{
        return d.bool(forKey: "kayitolusturulmusmu") 
    }
    func sifreKaydiOlusturulmusMuTrue(){
        d.set(true, forKey: "kayitolusturulmusmu")
    }
    func sifreKaydiOlusturulmusMuFalse(){
        d.set(false, forKey: "kayitolusturulmusmu")
    }
    func sifreKaydi(sifre:String){
        d.set(sifre, forKey: "sifre")
        d.set(true, forKey: "kayitolusturulmusmu")
    }
    func sifreSorgulama(dogrulanacakSifre:String)->Bool{
        if let alinanSifre = d.string(forKey: "sifre"){
            return dogrulanacakSifre.elementsEqual(alinanSifre)
        }else{
            return false
        }
    }
    func guvenlikSorusuKaydet(guvenlikSorusu:String){
        d.set(guvenlikSorusu, forKey: "guvenliksorusu")
    }
    func guvenlikSorusuGetir()->String{
        return d.string(forKey: "guvenliksorusu") ?? ""
    }
    func guvenlikSorusuCevapKaydet(guvenlikSorusuCevap:String){
        d.set(guvenlikSorusuCevap, forKey: "guvenliksorusucevap")
    }
    func guvenlikSorusuDogrula(dogrulanacakCevap:String)->Bool{
        if let kayitliCevap =  d.string(forKey: "guvenliksorusucevap"){
            return kayitliCevap.elementsEqual(dogrulanacakCevap)
        }else{
            return false
        }
    }
    func sifreSorulsunMuDegistir(sifreSorulsunMu:Bool){
        d.set(sifreSorulsunMu, forKey: "sifresorulsunmu")
    }
    func sifreDegistir(degistirilecekSifre:String){
        d.set(degistirilecekSifre, forKey: "sifre")
    }
    func kullaniciBilgileriniSifirla(){
        d.removeObject(forKey: "sifre")
        d.removeObject(forKey: "sifresorulsunmu")
        d.removeObject(forKey: "guvenliksorusu")
        d.removeObject(forKey: "kayitolusturulmusmu")
    }
}
